lib/bamboo/adapters/mailgun_adapter.ex

```diff
@@ -267,4 +268,10 @@ defmodule Bamboo.MailgunAdapter do
   end
 
   defp encode_body(body_without_attachments), do: Plug.Conn.Query.encode(body_without_attachments)
+
+  defp decode_body({:multipart, _} = multipart_body), do: multipart_body
+
+  defp decode_body(body) when is_binary(body) do
+    URI.decode_query(body)
```

Comment:
I was giving this PR another look, and I realized this is something of a counterpart to the existing `encode_body/1` function defined in line 270, right? 

What do you think about using [`Plug.Conn.Query.decode`](https://hexdocs.pm/plug/Plug.Conn.Query.html#decode/4) since we use the `Plug.Conn.Query.encode` function to encode the data?